bundles/binding/org.openhab.binding.sonance/src/main/java/org/openhab/binding/sonance/internal/SonanceBinding.java

```diff
@@ -40,541 +39,477 @@
+		String group = provider.getGroup(itemName);
+		String ip = provider.getIP(itemName);
+		int port = provider.getPort(itemName);
+
+		Socket s = null;
+		try {
+			s = new Socket(ip, port);
+			DataOutputStream outToServer = new DataOutputStream(s.getOutputStream());
+			BufferedReader i = new BufferedReader(new InputStreamReader(s.getInputStream()));
+
+			if (provider.isMute(itemName)) {
+				if (command.equals(OnOffType.OFF)) {
+					sendMuteCommand(itemName, SonanceConsts.MUTE_ON + group, outToServer, i);
+				} else if (command.equals(OnOffType.ON)) {
+					sendMuteCommand(itemName, SonanceConsts.MUTE_OFF + group, outToServer, i);
+				} else {
+					logger.error("I don't know what to do with the command \"{}\"", command);
+				}
+			} else if (provider.isPower(itemName)) {
+				if (command.equals(OnOffType.OFF)) {
+					sendPowerCommand(itemName, SonanceConsts.POWER_OFF, outToServer, i);
+				} else if (command.equals(OnOffType.ON)) {
+					sendPowerCommand(itemName, SonanceConsts.POWER_ON, outToServer, i);
+				} else {
+					logger.error("I don't know what to do with the command \"{}\"", command);
+				}
+			} else if (provider.isVolume(itemName))
+				if (command.equals(UpDownType.UP))
+					sendVolumeCommand(itemName, SonanceConsts.VOLUME_UP + group, outToServer, i);
+				else if (command.equals(UpDownType.DOWN))
+					sendVolumeCommand(itemName, SonanceConsts.VOLUME_DOWN + group, outToServer, i);
+				else {
+					try {
+						Double d = Double.parseDouble(command.toString());
+						setVolumeCommand(itemName, group, d.intValue(), outToServer, i, ip + ":" + port);
+					} catch (NumberFormatException nfe) {
+						logger.error("I don't know what to do with the volume command \"{}\" ({})", command,
+								nfe.getMessage());
+					}
+				}
+			s.close();
+		} catch (IOException e) {
+			logger.debug("IO Exception when sending command. Exception: {}", e.getMessage());
+		} finally {
+			closeSilently(s);
+		}
+	}
+
+	/**
+	 * Closes a socket
+	 * 
+	 * @param s
+	 *            socket to close
+	 */
+	private void closeSilently(Socket s) {
+		try {
+			if (s != null) {
+				s.close();
+			}
+		} catch (IOException e) {
+		}
+	}
+
+	/*
+	 * @see
+	 * org.openhab.core.binding.AbstractBinding#internalReceiveUpdate(java.lang
+	 * .String, org.openhab.core.types.State)
+	 */
+	@Override
+	protected void internalReceiveUpdate(String itemName, State newState) {
+		logger.info("Update received ({},{})", itemName, newState);
+
+		SonanceBindingProvider provider = findFirstMatchingBindingProvider(itemName);
+		String group = provider.getGroup(itemName);
+		String ip = provider.getIP(itemName);
+		int port = provider.getPort(itemName);
+
+		ip = null;
+		group = null; // cleanup
+
+		Socket s = null;
+		try {
+			s = new Socket(ip, port);
+			DataOutputStream outToServer = new DataOutputStream(s.getOutputStream());
+			BufferedReader i = new BufferedReader(new InputStreamReader(s.getInputStream()));
+
+			if (provider.isMute(itemName))
```

Comment:
Please use curly braces around every body, even one-liners (here and several below).